### PR TITLE
Post weekly performance stats in a Slack channel

### PR DESCRIPTION
**Pull request by felixclack**
We want to provide an easy way to track the performance metrics from
Find a Lost TRN.

It was decided that a message in Slack was the most appropriate way of
doing this.

We followed the lead of the teacher vacancies project to implement the scheduling
piece of this work with [sidekiq-cron](https://github.com/ondrejbartas/sidekiq-cron).

Right now, the message we post is very simple and only contains the
total number of TrnRequests created in the last week.

To complete setup of this, a Slack web hook URL is required,
ie. SLACK_WEBHOOK_URL.

This configuration variables allow us to run this in multiple environments and verify
it works.

### Checklist

- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: adr/00007-scheduled-jobs.md
Comment (tvararu) [inline adr/00007-scheduled-jobs.md:17]:
```suggestion
- run jobs via the infrastructure (they use sidekiq/sidekiq_cron)
```

Find-style is a relic of `Find postgraduate teacher training`.

File: adr/00007-scheduled-jobs.md
Comment (tvararu) [inline adr/00007-scheduled-jobs.md:22]:
```suggestion
and background processing properly, but using
```

**Review by tvararu — COMMENTED** [review]
Maybe instead of duplicating the content of the Apply scheduling ADR, we can just link to it and say "We will follow Apply's reasoning and use the `clockwork` gem: \<link\>"

File: app/services/slack_client.rb
Comment (tvararu) [inline app/services/slack_client.rb:25]: Can you add the `SLACK_TOKEN` environment variable to the terraform files and relay it to @merrilmathew who can add it to the environments?

File: config/clock.rb
Comment (tvararu) [inline config/clock.rb:12]: Should we only run this in the production hosting environment?

**Review by tvararu — COMMENTED** [review]
Before we merge this, can you test it by running the worker with a production Slack key from your local environment and @-ing me the resulting message on Slack? :+1:

**Comment by saliceti** [comment]
Could we please consider NOT using clock. It doesn't play well with terraform so much that we have to delete and recreate it at each deployment: https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/.github/actions/deploy/action.yml#L96

Teacher services have aligned on https://github.com/ondrejbartas/sidekiq-cron instead. It's easy to use (example: https://github.com/DFE-Digital/teaching-vacancies/blob/main/config/schedule.yml) and uses standard sidekiq workers

**Comment by tvararu** [comment]
@saliceti thanks for the suggestion!

File: app/services/slack_client.rb
Comment (tvararu) [inline app/services/slack_client.rb:15]: This uses the Slack API to post a message to a certain channel, which requires us to provision a Slack user with full read/write access.

There's a simpler way using the [Slack webhook API](https://api.slack.com/messaging/webhooks). It has less privileges; it only fires a message into a channel and can't read anything. Creating the Webhook API URL key should also be easier.

Here's how Apply does it: https://github.com/DFE-Digital/apply-for-teacher-training/blob/e85ff42219275d6728efabb225464340b195178d/app/workers/slack_notification_worker.rb#L7

Comment (tvararu) [reply app/services/slack_client.rb:15]: Also, we can just hardcode the `SLACK_CHANNEL` and not set it in an environment variable.

**Review by tvararu — CHANGES_REQUESTED** [review]

File: Gemfile
Comment (tvararu) [inline Gemfile:68]: Think this can be removed if we're not using Clockwork anymore?

File: app/services/slack_client.rb
Comment (tvararu) [inline app/services/slack_client.rb:11]:
```suggestion
    response = client.post('/', { text: message })
```

I think this is just a little bit clearer, wdyt?

**Review by tvararu — APPROVED** [review]
Smashing!!! :rocket:

**Pull request merged**
